src/common/types/column_data_collection.cpp

```diff
@@ -915,6 +915,9 @@ void ColumnDataCollection::Print() const {
 void ColumnDataCollection::Reset() {
 	count = 0;
 	segments.clear();
+
+	// Refreshes the ColumnDataAllocator to prevent holding on to allocated data unnecessarily
+	allocator = make_shared<ColumnDataAllocator>(*allocator);
```

Comment:
@lnkuiper would you mind double checking my fix here for the issue where the ColumnDataCollection  would not properly free data on reset?

Reply:
LGTM